### DISCUSSION TITLE
Getting an error while deploying solution to my subscription

### DISCUSSION THREAD

**Discussion by manojgan** [microsoft/PubSec-Info-Assistant#707] [Q&A]
I’d really appreciate any assistance in resolving the following issue.
I’m following the step-by-step instructions from the Deployment Guide.

When I run “make deploy” from the VSCode terminal window, I get the following error message…
-----------------------------------------------------------------------
Terraform has been successfully initialized!

You may now begin working with Terraform. Try running "terraform plan" to see
any changes that are required for your infrastructure. All Terraform commands
should now work.

If you ever set or change modules or backend configuration for Terraform,
rerun this command to reinitialize your working directory. If you forget, other
commands will detect it and remind you to do so if necessary.
module.bingSearch.data.template_file.workflow: Reading...
module.bingSearch.data.template_file.workflow: Read complete after 0s [id=0a65d2f94e2ffeb19f1ab5d616586e824d6a13d8006bf0b0b6f9f7b0ed590423]
╷
│ Error: building account: could not acquire access token to parse claims: running Azure CLI: exit status 1: ERROR: AADSTS530003: Your device is required to be managed to access this resource. Trace ID: 0cb9e830-fec4-43eb-8dc0-900d3aac7900 Correlation ID: b06c97e1-e810-4a4f-b7da-075c321c2547 Timestamp: 2024-05-14 12:49:23Z
│ Interactive authentication is needed. Please run:
│ az login --scope https://graph.microsoft.com/.default
│ 
│   with provider["registry.terraform.io/hashicorp/azurerm"],
│   on providers.tf line 23, in provider "azurerm":
│   23: provider "azurerm" {
----------------------------------------------------------------------
When I tried to run “az login --scope https://graph.microsoft.com/.default” from the terminal window after MFA authentication, it says “Trouble signing you. The request body must contain the following parameter" - client id”

Thanks in advance for the assistance.

**Comment by dayland** [comment]
This is due to running the Codespace in a browser tab combined with targeting an Azure Environment that has device policies enabled. The error message **Your device is required to be managed to access this resource.** is the key message. You will need to open the CodeSpace in VSCode or similar editor from a device that meets the policies required by your organization.

You can find more details at https://github.com/microsoft/PubSec-Info-Assistant/blob/main/docs/deployment/developing_in_a_GitHub_Codespaces.md and in particular on that page in the [Opening a codespace in VSCode](https://github.com/microsoft/PubSec-Info-Assistant/blob/main/docs/deployment/developing_in_a_GitHub_Codespaces.md#opening-a-codespace-in-vs-code) section.
